CONTRIBUTING.md

```diff
@@ -1,143 +1,266 @@
+you should follow.
+
+**Table of contents:**
+<!-- Please don't remove the following comments.
+To generate and update the TOC: https://github.com/mzlogin/vim-markdown-toc -->
+<!-- vim-markdown-toc GFM -->
+
+* [Use a Personal Fork and not Organization](#use-a-personal-fork-and-not-organization)
+* [Requirements](#requirements)
+* [Understanding Patches](#understanding-patches)
+* [Adding Patches](#adding-patches)
+* [Modifying Patches](#modifying-patches)
+  * [Method 1](#method-1)
+    * [Using the Paper tool](#using-the-paper-tool)
+    * [Manual method: Stashing](#manual-method-stashing)
+  * [Method 2 - Fixup commits](#method-2---fixup-commits)
+    * [Manual method](#manual-method)
+    * [Automatic method](#automatic-method)
+* [PR Policy](#pr-policy)
+* [Formatting](#formatting)
+* [Patch Notes](#patch-notes)
+* [Obfuscation Helpers](#obfuscation-helpers)
+* [Configuration files](#configuration-files)
+  * [PaperConfig example](#paperconfig-example)
+  * [PaperWorldConfig example](#paperworldconfig-example)
+* [Frequently Asked Questions](#frequently-asked-questions)
+  * [I can't find the NMS file I need!](#i-cant-find-the-nms-file-i-need)
+  * [Where can I learn what this method/field does?](#where-can-i-learn-what-this-methodfield-does)
+  * [My commit doesn't need a build, what do I do?](#my-commit-doesnt-need-a-build-what-do-i-do)
+  * [Patching and building is *really* slow, what can I do?](#patching-and-building-is-really-slow-what-can-i-do)
+
+<!-- vim-markdown-toc -->
 
 ## Use a Personal Fork and not Organization
-Paper will routinely modify your PR for you to rebase it on latest or take care of any minor nitpicks that we can
-just solve for you instead of making you go back for a trivial item.
 
-However, if you use an Organization for the PR, this prevents Paper from doing this, and will require us to manually
-merge it and close the PR instead of it being marked as the purple merged.
+Paper will routinely modify your PR for you to rebase it on latest or take care
+of any minor nitpicks that we can just solve for you instead of making you go
+back for a trivial item.
 
-We much prefer to have merged PR's show as merged, so please do not use organizations repo's for PR as GitHub sadly
-does not support maintainers pushing to them.
+However, if you use an Organization for the PR, this prevents Paper from doing
+this, and will require us to manually merge it and close the PR instead of it
+being marked as the purple merged.
 
-See [https://github.com/isaacs/github/issues/1681](https://github.com/isaacs/github/issues/1681)
+We much prefer to have merged PR's show as merged, so please do not use
+organizations repositories for PR as GitHub sadly does not support maintainers
+pushing to them.
 
-## Understanding Patches
-Patches to Paper are very simple, but center around the directories 'Paper-API' and 'Paper-Server'
+See <https://github.com/isaacs/github/issues/1681> for more information on the
+issue.
```

Comment:
```suggestion
```

Reply:
The link was present in the original document, and provides reasoning for the requirement. If someone sees it when it's closed, we can also remove this section from the file altogether.